framework/wazuh/core/indexer/events.py

```diff
@@ -1,29 +1,57 @@
+import asyncio
 from dataclasses import asdict
+from typing import List
+from uuid import UUID
 
 from .base import BaseIndex
 from wazuh.core.indexer.models.events import Events
+from wazuh.core.indexer.bulk import MixinBulk
 
+from opensearchpy import AsyncOpenSearch
 
-class EventsIndex(BaseIndex):
+
+HARDCODED_EVENTS_INDEX_NAME = "events"
+
+
+class EventsIndex(BaseIndex, MixinBulk):
     """Set of methods to interact with the stateful events indices."""
+    INDEX = HARDCODED_EVENTS_INDEX_NAME
+
+    def __init__(self, client: AsyncOpenSearch):
+        super().__init__(client)
 
-    async def create(self, events: Events) -> dict:
+    async def create(self, events: Events, batcher_client) -> dict:
         """Post new events to the indexer.
 
         Parameters
         ----------
         events : Events
             List of events.
-        
+
         Returns
         -------
         dict
             Indexer response for each one of the events.
         """
-        # TODO(#24713): Implement server to indexer events batching
+        list_of_uid: List[UUID] = []
         response = {}
-        for i, event in enumerate(events.events):
-            resp = await self._client.create(index=event.get_index_name(), body=asdict(event))
-            response.update({f'{i}': resp})
+
+        # Sends the events to the batcher
+        for event in events.events:
+            uid_of_request = batcher_client.send_event(asdict(event))
+            list_of_uid.append(uid_of_request)
+
+        # Create tasks using a lambda function to obtain the result of each one
+        tasks = []
+        for uid in list_of_uid:
+            task = asyncio.create_task(
+                (lambda u: batcher_client.get_response(u))(uid)
+            )
+            tasks.append(task)
+
+        # Wait for all of them and create response
+        results = await asyncio.gather(*tasks)
+        for i, result in enumerate(results):
+            response.update({f'{i}': result})
```

Comment:
Like I mentioned in a comment above, the response shouldn't be formatted with `response.update({f'{i}': result})`

```suggestion
        # Sends the events to the batcher
        for event in events.events:
            request_id = batcher_client.send_event(asdict(event))
            ids.append(request_id)

        # Create tasks using a lambda function to obtain the result of each one
        tasks = []
        for id in ids:
            task = asyncio.create_task(
                (lambda u: batcher_client.get_response(u))(id)
            )
            tasks.append(task)

        # Wait for all of them and create response
        results = await asyncio.gather(*tasks)
        for i, result in enumerate(results):
            response.update({f'{i}': result})
```

Reply:
Changed in 92c3106a91132f0a82fbc5b27062532886153f28. Example response:

```json
{
    "events": [
        {
            "id": "aSp0mZEBKT6pxgp_hmy6",
            "result": "created",
            "status": 201
        },
        {
            "id": "aip0mZEBKT6pxgp_hmy6",
            "result": "created",
            "status": 201
        },
        {
            "id": "ayp0mZEBKT6pxgp_hmy6",
            "result": "created",
            "status": 201
        },
        {
            "id": "bCp0mZEBKT6pxgp_hmy6",
            "result": "created",
            "status": 201
        },
        {
            "id": "bSp0mZEBKT6pxgp_hmy6",
            "result": "created",
            "status": 201
        },
        {
            "id": "bip0mZEBKT6pxgp_hmzB",
            "result": "created",
            "status": 201
        },
        {
            "id": "byp0mZEBKT6pxgp_hmzB",
            "result": "created",
            "status": 201
        },
        {
            "id": "cCp0mZEBKT6pxgp_hmzB",
            "result": "created",
            "status": 201
        },
        {
            "id": "cSp0mZEBKT6pxgp_hmzB",
            "result": "created",
            "status": 201
        },
        {
            "id": "cip0mZEBKT6pxgp_hmzB",
            "result": "created",
            "status": 201
        },
        {
            "id": "cyp0mZEBKT6pxgp_hmzM",
            "result": "created",
            "status": 201
        },
        {
            "id": "dCp0mZEBKT6pxgp_hmzM",
            "result": "created",
            "status": 201
        },
        {
            "id": "dSp0mZEBKT6pxgp_hmzM",
            "result": "created",
            "status": 201
        },
        {
            "id": "dip0mZEBKT6pxgp_hmzM",
            "result": "created",
            "status": 201
        },
        {
            "id": "dyp0mZEBKT6pxgp_hmzM",
            "result": "created",
            "status": 201
        },
        {
            "id": "eCp0mZEBKT6pxgp_hmzV",
            "result": "created",
            "status": 201
        },
        {
            "id": "eSp0mZEBKT6pxgp_hmzV",
            "result": "created",
            "status": 201
        },
        {
            "id": "eip0mZEBKT6pxgp_hmzV",
            "result": "created",
            "status": 201
        },
        {
            "id": "eyp0mZEBKT6pxgp_hmzV",
            "result": "created",
            "status": 201
        },
        {
            "id": "fCp0mZEBKT6pxgp_hmzV",
            "result": "created",
            "status": 201
        },
        {
            "id": "fSp0mZEBKT6pxgp_hmzg",
            "result": "created",
            "status": 201
        },
        {
            "id": "fip0mZEBKT6pxgp_hmzg",
            "result": "created",
            "status": 201
        },
        {
            "id": "fyp0mZEBKT6pxgp_hmzg",
            "result": "created",
            "status": 201
        },
        {
            "id": "gCp0mZEBKT6pxgp_hmzg",
            "result": "created",
            "status": 201
        },
        {
            "id": "gSp0mZEBKT6pxgp_hmzg",
            "result": "created",
            "status": 201
        }
    ]
}
```